src/smc-webapp/account/actions.ts

```diff
@@ -0,0 +1,354 @@
+    const err_help = `\
+Please reload this browser tab and try again.
+
+If that doesn't work after a few minutes, try these ${doc_conn} or email ${this.help()}.\
+`;
+
+    this.setState({ signing_in: true });
+    webapp_client.sign_in({
+      email_address: email,
+      password,
+      remember_me: true,
+      timeout: 30,
+      get_api_key: this.redux.getStore("page").get("get_api_key"),
+      cb: (error, mesg) => {
+        this.setState({ signing_in: false });
+        if (error) {
+          this.setState({
+            sign_in_error: `There was an error signing you in (${error}). ${err_help}`
+          });
+          return;
+        }
+        switch (mesg.event) {
+          case "sign_in_failed":
+            this.setState({ sign_in_error: mesg.reason });
+            return;
+          case "signed_in":
+            //redux.getActions('page').set_active_tab('projects')
+            break;
+          case "error":
+            this.setState({ sign_in_error: mesg.reason });
+            return;
+          default:
+            // should never ever happen
+            this.setState({
+              sign_in_error: `The server responded with invalid message when signing in: ${JSON.stringify(
+                mesg
+              )}`
+            });
+            return;
+        }
+      }
+    });
+  }
+
+  create_account(first_name, last_name, email, password, token, usage_intent?) {
+    this.setState({ signing_up: true });
+    webapp_client.create_account({
+      first_name,
+      last_name,
+      email_address: email,
+      password,
+      usage_intent,
+      agreed_to_terms: true,
+      token,
+      get_api_key: this.redux.getStore("page").get("get_api_key"),
+      cb: (err, mesg) => {
+        this.setState({ signing_up: false });
+        if (err != null) {
+          // generic error.
+          this.setState({ sign_up_error: { generic: JSON.stringify(err) } });
+          return;
+        }
+        switch (mesg.event) {
+          case "account_creation_failed":
+            this.setState({ sign_up_error: mesg.reason });
+            return;
+          case "signed_in":
+            this.redux.getActions("page").set_active_tab("projects");
+            var { analytics_event, track_conversion } = require("../misc_page");
+            analytics_event("account", "create_account"); // user created an account
+            track_conversion("create_account");
+            return;
+          default:
+          // should never ever happen
+          // alert_message(type:"error", message: "The server responded with invalid message to account creation request: #{JSON.stringify(mesg)}")
+        }
+      }
+    });
+  }
+  // deletes the account and then signs out everywhere
+  delete_account() {
+    async.series(
+      [
+        async cb => {
+          // cancel any subscriptions
+          try {
+            await this.redux.getActions("billing").cancel_everything();
+            return cb();
```

Comment:
return?

Reply:
No idea. Coffee code would return here right? So I didn't touch it to be safer.

Reply:
No.  I think there is no situation anywhere ever that the value of calling a callback function gets returned.  Definitely not in the async library. This is just one of those stupid side effects of coffeescript.